### PR TITLE
Make the env fix cross platform

### PR DESCRIPTION
**Pull request by steelbrain**
Just like the title says

### CODE REVIEW

**Comment by steelbrain** [comment]
All tests green via apm

**Comment by AsaAyers** [comment]
How exactly does this change any behavior?

I see you dropped `process.env.hasOwnProperty(prop)`, but this only matters if the object you're iterating over extends another object that might have properties. As long as `process.env.prototype is undefined` then checking `hasOwnProperty` is going to be `true` for all properties.

fyi, CoffeeScript will let you add a `when` to for loops. This loop is exactly the same as both of the previous implementations.
```CoffeeScript
for prop,value of process.env when prop isnt 'OS'
  options.env[prop] = value 
```

File: lib/helpers.coffee [hunk 1]
```diff
@@ -32,9 +32,8 @@ module.exports = Helpers =
           resolve(data.stderr.join(''))
       if isNodeExecutable
         options.env ?= {}
-        for prop in process.env
-          if process.env.hasOwnProperty(prop) and prop isnt 'OS'
-            options.env[prop] = process.env[prop]
+        for prop,value of process.env
```
Comment (iam4x) [inline lib/helpers.coffee]: `for prop, value`

**Comment by iam4x** [comment]
And +1 for the `when` keyword

File: lib/helpers.coffee (hunk 1 above)
Comment (AsaAyers) [inline lib/helpers.coffee]: Thanks I missed the change from `for ... in` to `for ... of` because it was hidden inside all the formatting changes.

Comment (steelbrain) [inline lib/helpers.coffee]: Sorry guys, was low on caffeine. :sweat_smile:

Comment (steelbrain) [inline lib/helpers.coffee]: and to @AsaAyers I know it's quite confusing, but it wasn't working for me and some other people that way, and the way it was enumerating it always resulted the env to be `{}` on linux and a valid object on windows. I know it's super weird, but this style change makes it work everywhere. I have confirmed it on a few machines, both before and after.

Comment (AsaAyers) [inline lib/helpers.coffee]: You could have simply changed the `for ... in` to be `for ... of` to fix it. None of the changes you made were bad, it's just that you have a very small fix mixed in with enough reformatting that I couldn't find the actual fix.

Now that I see the functional change I don't doubt it fixed it.

**Pull request merged**